src/Compilers/CSharp/Portable/Parser/Lexer.cs

```diff
@@ -444,7 +444,26 @@ private void ScanSyntaxToken(ref TokenInfo info)
                     if (!this.ScanNumericLiteral(ref info))
                     {
                         TextWindow.AdvanceChar();
-                        info.Kind = SyntaxKind.DotToken;
+                        if (TextWindow.PeekChar() == '.')
+                        {
+                            TextWindow.AdvanceChar();
+                            if (TextWindow.PeekChar() == '.')
+                            {
+                                // Triple-dot: explicitly reject this, to allow triple-dot
+                                // to be added to the language without a breaking change.
+                                // (without this, 0...2 would parse as (0)..(.2), i.e. a range from 0 to 0.2)
+                                TextWindow.Reset(startingPosition);
+                                goto default;
```

Comment:
>goto default [](start = 32, length = 12)

This will result in a very confusing "unexpected character: '.'" on the first character. I think just scan it as a range token with an error on it.